### PR TITLE
Breaking: Pinocchio v2.0.0

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

This PR updates sot-dynamic-pinocchio for pinocchio v2.0.0:
- change the `se3` c++ namespace to `pinocchio`
- change the `se3` python namespace to `pin` (as it is advised in the [porting](https://github.com/stack-of-tasks/pinocchio/blob/devel/doc/_porting.md) part of the documentation)
- remove an include
- update the CMake dependency

:warning: These changes will drop compatibility with Pinocchio < 2.0.0 :warning: 

ref https://github.com/stack-of-tasks/pinocchio/issues/570

### CODE REVIEW

File: src/sot-dynamic-pinocchio.cpp [hunk 1]
```diff
@@ -754,15 +754,15 @@ computeGenericJacobian(const bool isFrame, const int jointId, dg::Matrix& res,co
...
   //TODO: Find a way to remove tmp object
-  se3::Data::Matrix6x tmp = Eigen::MatrixXd::Zero(6,m_model->nv);
+  pinocchio::Data::Matrix6x tmp = Eigen::MatrixXd::Zero(6,m_model->nv);
 
   //Computes Jacobian in world coordinates.
   if(isFrame){
-    se3::getJacobian<se3::WORLD>(*m_model,*m_data,
-                                 m_model->frames[(se3::Model::Index)jointId].parent,tmp);
+    pinocchio::getJacobian<pinocchio::WORLD>(*m_model,*m_data,
+                                 m_model->frames[(pinocchio::Model::Index)jointId].parent,tmp);
```
Comment (jmirabel) [inline src/sot-dynamic-pinocchio.cpp]: getJacobian became deprecated. Can you change it to
```cpp
getJointJacobian(*m_model,*m_data,
                            m_model->frames[(pinocchio::Model::Index)jointId].parent
                            pinocchio::WORLD,
                            tmp)
```

**Comment by jmirabel** [comment]
Thanks. This looks good to me.

**Comment by nim65s** [comment]
Well… This broke the build for me :/
Please don't merge :)

**Comment by jmirabel** [comment]
> Well… This broke the build for me :/
> Please don't merge :)

Yes, I didn't check carrefully :)

**Comment by nim65s** [comment]
This works now, thanks for the review and the heads-up, @jmirabel :)

**Pull request merged**